cpp/include/cudf/types.hpp

```diff
@@ -272,6 +272,20 @@ class data_type {
  */
 inline bool operator==(data_type const& lhs, data_type const& rhs) { return lhs.id() == rhs.id(); }
 
+/**
+ * @brief Compares two `data_type` objects for inequality.
+ *
+ * // TODO Define exactly what it means for two `data_type`s to be equal. e.g.,
```

Comment:
Is this TODO about equality still an "unresolved" part of the data type design? I added this `!=` operator for use in the linearizer code. I could remove this and put in a lambda function where I use it, but it was nice to be able to pass `std::not_equal_to<>()` to `std::adjacent_find`.